marketdata/market/coinmarketcap/cmc.go

```diff
@@ -0,0 +1,80 @@
+package cmc
+
+import (
+	"github.com/spf13/viper"
+	"github.com/trustwallet/blockatlas/marketdata/market"
+	"github.com/trustwallet/blockatlas/pkg/blockatlas"
+	"github.com/trustwallet/blockatlas/pkg/logger"
+	"math/big"
+	"net/url"
+	"time"
+)
+
+type Market struct {
+	market.Market
+}
+
+func InitMarket() market.Provider {
+	m := &Market{
+		Market: market.Market{
+			Id:         "cmc",
+			Name:       "CoinMarketCap",
+			URL:        "https://coinmarketcap.com/",
+			Request:    blockatlas.InitClient(viper.GetString("market.cmc_api")),
+			UpdateTime: time.Second * 30,
+		},
+	}
+	m.Headers["X-CMC_PRO_API_KEY"] = viper.GetString("market.cmc_api_key")
+	return m
+}
+
+func (m *Market) GetData() (blockatlas.Tickers, error) {
+	var prices CoinPrices
+	err := m.Get(&prices, "v1/cryptocurrency/listings/latest", url.Values{"limit": {"5000"}, "convert": {"USD"}})
```

Comment:
what do you think of moving network related code to separate client.go as it's been done for blockchain classes?

Reply:
it's a lot layer for a simple thing. Make sense if the method getData have a lot of lines